### ISSUE TITLE
Using webgl's filter() with lighting enabled does not work

### ISSUE THREAD

**Issue by inaridarkfox4231** [processing/p5.js#6641]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

1.9.0

### Web browser and version

Chrome

### Operating System

Windows11

### Steps:
1. Draw a webgl sketch that uses lighting
2. Use filter().
3. A solid square is drawn. It's a bug.

## no lights
```js
function setup() {
  createCanvas(400, 400, WEBGL);
}

function draw() {
  background(128);
  noStroke();
  fill("red");
  sphere(40);

  filter(BLUR, 4);
}
```

![バグサンプル1](https://github.com/processing/p5.js/assets/39549290/35e56a31-90be-4e47-b4b3-e312e47974fd)

## use lights
```js
function setup() {
  createCanvas(400, 400, WEBGL);
}

function draw() {
  background(128);
  lights();
  noStroke();
  fill("red");
  sphere(40);

  filter(BLUR, 4);
}
```

![バグサンプル2](https://github.com/processing/p5.js/assets/39549290/fcab1dc4-1f79-4341-b1e6-6ca4406208c2)

I'll add a few things, because I don't want a sloppy solution due to lack of explanation.

### First, this behavior occurs with all eight default filters. This example uses BLUR, but it's not essential.
Also, lights() is calling multiple lights, but it seems to occur even if it is a single light.

```js
function setup() {
  createCanvas(400, 400, WEBGL);
}

function draw() {
  background(128);
  ambientLight(64);
  noStroke();
  fill("red");
  sphere(40);

  filter(BLUR, 4);
}
```

![バグサンプル3](https://github.com/processing/p5.js/assets/39549290/3340593a-3e48-4735-9028-dd4df610d13b)

It got dark like this. It seems that it is affected by the light. Maybe noLights() is not used properly...?

There seems to be no problem when drawing in 2D or using webgl without a light.

![バグサンプル4](https://github.com/processing/p5.js/assets/39549290/de66be1a-e0bf-4f5b-81c3-dfbd708231d7)

This problem seems to occur only when using lights with webgl.

I can't work on this problem. I'm not involved in developing the filter specifications, so I don't know how to resolve this issue. I'm sure the people involved in developing the specifications are familiar with the filter specifications, so I hope they can resolve it.

**Comment by inaridarkfox4231** [comment]
I'm not sure if users are looking for a specification where the filter functions properly even when lighting is enabled, but I think it's at least more natural that way.

https://github.com/processing/p5.js/assets/39549290/0adf56b7-d82d-4060-aa4c-38a086b7e554

**Comment by inaridarkfox4231** [comment]
```js
function setup() {
  createCanvas(400, 400, WEBGL);
}

function draw() {
  background(128);
  lights();
  noStroke();
  fill("red");
  rotateX(PI/3);
  rotateY(PI/3);
  box(200);

  noLights();

  filter(BLUR, 4);
}
```

![バグサンプル6](https://github.com/processing/p5.js/assets/39549290/8b11e60a-19fc-47e8-abca-78817d5d0398)

It seems that just adding noLights() results in the intended behavior. Looks like I need to add it somewhere.

**Comment by inaridarkfox4231** [comment]
Apparently all we have to do is add noLights() at the very beginning of webgl's filter function. It may affect other areas, but I'm not sure since I'm not familiar with p5.Framebuffer.

![解決？](https://github.com/processing/p5.js/assets/39549290/92d373c5-e15a-4ee3-b36f-9b9d9d352a94)

**Comment by inaridarkfox4231** [comment]
Seems to be resolved in #6595. excuse me.
